src/bosh_azure_cpi/lib/cloud/azure/vms/instance_type_mapper.rb

```diff
@@ -4,95 +4,88 @@ module Bosh::AzureCloud
-      %w[Standard_DS11_v2 Standard_DS12_v2 Standard_DS13_v2 Standard_DS14_v2 Standard_DS15_v2],
-      %w[Standard_B1s Standard_B1ms Standard_B2s Standard_B2ms Standard_B4ms Standard_B8ms],
-      %w[Standard_F1s Standard_F2s Standard_F4s Standard_F8s Standard_F16s],
-      %w[Standard_GS1 Standard_GS2 Standard_GS3 Standard_GS4 Standard_GS5],
-      %w[Standard_DS1 Standard_DS2 Standard_DS3 Standard_DS4 Standard_DS11 Standard_DS12 Standard_DS13 Standard_DS14],
-      # Standard Storage
-      %w[Standard_D2_v3 Standard_D4_v3 Standard_D8_v3 Standard_D16_v3 Standard_D32_v3 Standard_D64_v3],
-      %w[Standard_E2_v3 Standard_E4_v3 Standard_E8_v3 Standard_E16_v3 Standard_E32_v3 Standard_E64_v3 Standard_E64i_v3],
-      %w[Standard_D1_v2 Standard_D2_v2 Standard_D3_v2 Standard_D4_v2 Standard_D5_v2],
-      %w[Standard_A1_v2 Standard_A2_v2 Standard_A4_v2 Standard_A8_v2 Standard_A2m_v2 Standard_A4m_v2 Standard_A8m_v2],
-      %w[Standard_D11_v2 Standard_D12_v2 Standard_D13_v2 Standard_D14_v2 Standard_D15_v2],
-      %w[Standard_F1 Standard_F2 Standard_F4 Standard_F8 Standard_F16],
-      %w[Standard_G1 Standard_G2 Standard_G3 Standard_G4 Standard_G5],
-      %w[Standard_D1 Standard_D2 Standard_D3 Standard_D4 Standard_D11 Standard_D12 Standard_D13 Standard_D14],
-      %w[Standard_A1 Standard_A2 Standard_A3 Standard_A4 Standard_A5 Standard_A6 Standard_A7 Standard_A8 Standard_A9 Standard_A10 Standard_A11],
-      # High performance compute
-      %w[Standard_H8 Standard_H16 Standard_H8m Standard_H16m Standard_H16r Standard_H16mr]
-    ].freeze
-
-    def map(desired_instance_size, available_vm_sizes)
+    SERIES_PREFERENCE = {
+      'D' => 5,  # General purpose - balanced
+      'F' => 4,  # Compute optimized
+      'E' => 4,  # Memory optimized
+      'B' => 3,  # Burstable VMs
+      'L' => 3,  # Storage optimized
+      'A' => 2,  # Basic general purpose
+      'H' => 1,  # High performance compute
+      'M' => 1,  # Ultra memory optimized
+      'N' => 1   # GPU
+    }.freeze
+
+    def initialize(azure_client)
+      @azure_client = azure_client
       @logger = Bosh::Clouds::Config.logger
-      @logger.debug("The available VM sizes in the specified region are '#{_vm_sizes_to_string(available_vm_sizes)}'")
-      possible_vm_sizes = _find_possible_vm_sizes(desired_instance_size, available_vm_sizes)
-      cloud_error("Unable to meet desired instance size: #{desired_instance_size['cpu']} CPU, #{desired_instance_size['ram']} MB RAM") if possible_vm_sizes.empty?
-      @logger.debug("The possible VM sizes which meet desired instance size are '#{_vm_sizes_to_string(possible_vm_sizes)}'")
+      @sku_cache = {}
+    end
 
-      closest_matched_vm_sizes = _find_closest_matched_vm_sizes(possible_vm_sizes)
-      @logger.debug("The closest matched VM sizes are '#{closest_matched_vm_sizes}'")
+    def map(desired_instance_size, location)
```

Comment:
It's hard to evaluate, but I assume these will be breaking changes since it's hard to guarantee that users will get the same VM type match they used to.

Reply:
Absolutely, these changes are breaking and should be released as part of a major version update and mentioned in the release notes. I added a notice to the changelog section in the PR body.